configsuite/schema.py

```diff
@@ -86,54 +95,107 @@ def _check_required_not_default(schema_level):
+        MK.Required: {
+            MK.Type: types.Bool,
+            MK.Required: False,
+            MK.Default: _SCHEMA_LEVEL_DEFAULTS[MK.Required],
+        },
+        MK.AllowNone: {
+            MK.Type: types.Bool,
+            MK.Required: False,
+            MK.Default: _SCHEMA_LEVEL_DEFAULTS[MK.AllowNone],
+        },
+        MK.Description: {
+            MK.Type: types.String,
+            MK.Required: False,
+            MK.Default: _SCHEMA_LEVEL_DEFAULTS[MK.Description],
+        },
+        MK.Default: {MK.Type: _Anytype, MK.Required: False, MK.AllowNone: True},
         MK.ElementValidators: {
             MK.Type: types.List,
-            MK.Required: False,
             MK.Content: {MK.Item: {MK.Type: types.Callable}},
         },
         MK.ContextValidators: {
             MK.Type: types.List,
-            MK.Required: False,
             MK.Content: {MK.Item: {MK.Type: types.Callable}},
         },
-        MK.Transformation: {MK.Type: types.Callable, MK.Required: False},
-        MK.ContextTransformation: {MK.Type: types.Callable, MK.Required: False},
-        MK.LayerTransformation: {MK.Type: types.Callable, MK.Required: False},
+        MK.Transformation: {
+            MK.Type: types.Callable,
+            MK.Required: False,
+            MK.AllowNone: True,
+        },
+        MK.ContextTransformation: {
+            MK.Type: types.Callable,
+            MK.Required: False,
+            MK.AllowNone: True,
+        },
+        MK.LayerTransformation: {
+            MK.Type: types.Callable,
+            MK.Required: False,
+            MK.AllowNone: True,
+        },
     },
 }
 
 
-def assert_valid_schema(schema, allow_default=False):
-    assert_valid_schema_level(schema, allow_default=allow_default)
+def assert_valid_schema(schema):
+    _assert_valid_schema(
+        _META_SCHEMA, allow_default=False, validate_named_keys=False
+    )
+
+    _assert_valid_schema(schema, allow_default=False, validate_named_keys=True)
+
+
+def _assert_valid_schema(schema, allow_default, validate_named_keys):
+    _assert_valid_schema_level(schema, allow_default)
 
     level_type = schema[MK.Type]
     if isinstance(level_type, types.BasicType):
         return
     elif level_type == types.NamedDict:
-        _assert_valid_named_dict_schema(schema)
+        _assert_valid_named_dict_schema(schema, validate_named_keys)
     elif level_type == types.List:
-        _assert_valid_list_schema(schema)
+        _assert_valid_list_schema(schema, validate_named_keys)
     elif level_type == types.Dict:
-        _assert_valid_dict_schema(schema)
+        _assert_valid_dict_schema(schema, validate_named_keys)
     else:
         raise TypeError("Unknown base container: {}".format(schema))
 
 
-def assert_valid_schema_level(schema, allow_default):
-    level_schema = copy.deepcopy(schema)
-    if MK.Content in level_schema:
-        level_schema.pop(MK.Content)
+def _build_level_schema(schema):
```

Comment:
The schema logic got too complicated, so extracted it our in a separate function...